lib/Date/Language/French.pm

```diff
@@ -32,5 +32,6 @@ sub format_b { $MoYs[$_[0]->[4]] }
 sub format_B { $MoY[$_[0]->[4]] }
 sub format_h { $MoYs[$_[0]->[4]] }
 sub format_p { $_[0]->[2] >= 12 ?  $AMPM[1] : $AMPM[0] }
+sub format_o { sprintf("%2de",$_[0]->[3]) }
```

Comment:
a better French fix will be to do not use the suffix as it s not very common

The current suggested fix will be wrong in several cases
for example: `1st` should be `1er, 
then after `ème` is better than `e` for every other enumeration ....

Reply:
Not being French myself, I defer to you and (wikipedia)[https://en.wikipedia.org/wiki/Date_and_time_notation_in_France], the better fix is do not use the suffix.

Will update the pull request accordingly